android/titanium/src/java/org/appcelerator/titanium/view/TiUIFragment.java

```diff
@@ -44,12 +49,49 @@ public boolean dispatchTouchEvent(MotionEvent ev)
 
 			FragmentManager manager = ((FragmentActivity) activity).getSupportFragmentManager();
 			FragmentTransaction transaction = manager.beginTransaction();
+			transaction.runOnCommit(onCommitRunnable);
 			fragment = createFragment();
 			transaction.add(container.getId(), fragment);
 			transaction.commitAllowingStateLoss();
 		}
 	}
 
+	private Runnable onCommitRunnable = new Runnable() {
+		@Override
+		public void run()
+		{
+			// Draw all the children that have been added prior the fragment transaction commit.
+			realizeFragmentViews();
+			fragmentCommitted = true;
+		}
+	};
+
+	public void realizeFragmentViews()
+	{
+		for (TiUIView tiUIView : childrenToRealize) {
+			// Draw the views
+			((ViewGroup) getNativeView()).addView(tiUIView.getOuterView(), tiUIView.getLayoutParams());
+			// Move them to the default children array
+			children.add(tiUIView);
+			// Clear and nullify the childrenToRealize array
+			childrenToRealize.clear();
+			childrenToRealize = null;
+		}
+	}
+
+	@Override
+	public void add(TiUIView child)
+	{
+		// If the fragment transaction has been committed add the children the usual way
+		if (fragmentCommitted) {
+			super.add(child);
+		} else {
+			// If the fragment has not been added to the native view add the children in
+			// the array to be realized
+			childrenToRealize.add(child);
+		}
+	}
+
```

Comment:
@ypbnv, we also need to override the `insertAt(TiUIView, int)` and `remove(TiUIView)` methods. It's an unlikely edge-case, but it should be easy to add support, no?

For example, see our `TiUIScrollView` code here...
https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java#L1065

Reply:
Yes, I thought about `insertAt()`, but what I found out took the focus at that moment. I have described it in here: 
https://jira.appcelerator.org/browse/TIMOB-26283
Let me know what Gary and you think about that. If `insertAt()` needs a fix it will be better to do it for the source before overriding it here. 

I will override `remove()` , too.